<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера V.2</h3>
    
Удачи в следующих споринтах, небольшой комментарий про `transform` оставил</div>  

# <div class="alert alert-success">
<h2> Комментарий ревьюера</h2>
    
Привет! Меня зовут Алексей Тюрин, я буду делать ревью твоей работы. Давай будем общаться на «ты».

У тебя хорошая, аккуратная работа 👍. В ней всё здорово, я её вернул, чтобы у тебя была возможность посмотреть комментарии и задать вопросы, если есть. Если вопросов нет, то просто зашли работу обратно и я приму :)

---
Комментарии делятся на 3 группы:
    
Зелёным цветом отмечены комментарии, либо подчёркивающие удачные решения, либо содержащие какую-то дополнительную информацию или советы, они формально не требуют правки, но правки приветствуются. Выделены они так:

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Тут текст    
</div>    

Жёлтым цветом выделены правки, которые желательны:

<div class="alert alert-warning"; style="border-left: 7px solid yellow" >
<h3>!! Комментарий ревьюера</h3>
    
Тут текст    
</div>    

Красным выделены правки, которые обязательны
    
<div class="alert alert-danger"; style="border-left: 7px solid red">
<h3>👀 Комментарий ревьюера</h3>
    
Тут текст    
</div>    
    
Давай работать над проектом в диалоге: если ты **что-то меняешь** в проекте по моим рекомендациям — **пиши об этом**. Выбери для своих комментариев какой-то заметный цвет, так мне будет легче отследить изменения. Пожалуйста, **не перемещай, не изменяй и не удаляй мои комментарии**. Всё это поможет выполнить повторную проверку твоего проекта оперативнее. Если оставляешь комментарии на какой-то итерации ревью, пиши что-то вроде `комментарий студента V.{номер итерации}`
</div>

Цвет можно сделать, например, так

<div class = "alert alert-info" style="border-left: 7px solid blue">
<h3>Комментарий студента</h3>
Ревьюер душнила :)
</div>

Или так:
    
<div class="alert alert-info" style="background:#ffdbf1;color:#2e00ab;border-left:7px solid #2e00ab">
<h3>Комментарий студента </h3>
    ...
</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [80]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [81]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [82]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [83]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [84]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Я знаю, что это код из автоматического решения, но напишу тут пару слов.

С точки зрения кода такую замену можно сделать через `transform`: `data['total_income'] = data['total_income'].fillna(data.groupby('income_type')['total_income'].transform(lambda x : x.median()))`. `transform` в материалах спринта нет, но это полезный метод. Он намного лучше масштабируется на бОльшее число групп, чем такой цикл.
    
Такая замена работает, только если пропуски распределены случайно, т.е. вероятность пропуска в строке не зависит от других значений в строке. Что если бы все прпоуски в столбце `total_income` были только у людей без образования, а у тех людей без образования, где пропусков нет, доход указан 0? Тогда надо было бы заменить пропуски на 0.
    
Проверять такое можно по-разному, можно, например, для каждого столбца сравнить распределение значения для среза с прпоусками в `total_income` и среза без пропусков. Сделать это можно через `value_counts(normalize=True)`, либо можно группировать по каждому столбцу и смотреть долю пропусков в каждой группе, если их одинаково пропуски распределены случайно (по крайней мере относительно выбранного столбца). 
    
Подробнее об этом можно почитать тут https://loginom.ru/blog/missing  или посмотреть видео: https://youtu.be/YpqUbirqFxQ
</div>    


<div class = "alert alert-info" style="border-left: 7px solid blue">
<h3>✔ Комментарий студента 1 </h3>
    Алексей, большое спасибо! 
    Только мне не очень понятно, что означает запись в скобках .transform(lambda x : x.median()). Где можно более подробно почитать про метод .tranform()? Я погуглила, но, к сожалению, с синтаксисом и преимуществами метода 
не разобралась.
    Заранее спасибо!
</div>  

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера V.2</h3>
    
Проблема в том, что особо не нагуглить :) Кажется, что лучше, чем https://stackoverflow.com/questions/27517425/apply-vs-transform-on-a-group-object ничего нет
        
Преимущества в том, что такой способ лучше масштабируется и гаранитирует, что в результате получится серия нужного размера
</div>  

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [85]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [86]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [87]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [88]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [89]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [90]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [91]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [92]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [93]:
data.duplicated().sum()

54

In [94]:
data = data.drop_duplicates()

**Задание 15. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [95]:
data['education'] = data['education'].str.lower()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [96]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [97]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [98]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [99]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [100]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследуйте данные и ответьте на вопросы

**Задание 19. Есть ли зависимость между количеством детей и возвратом кредита в срок?**

In [101]:
grouped_debt_children = data.groupby('children').agg({'debt':['sum','count']}) #sum показывает количество должников, count - общее число
#print(grouped_debt_children)
grouped_debt_children['percent_debt'] = grouped_debt_children['debt']['sum']/grouped_debt_children['debt']['count']*100 #процент должников в зависимости от числа детей

print(grouped_debt_children)

          debt        percent_debt
           sum  count             
children                          
0         1063  14107     7.535266
1          444   4809     9.232689
2          194   2052     9.454191
3           27    330     8.181818
4            4     41     9.756098
5            0      9     0.000000


**Вывод:** Процент должников не превышает 10% как для людей без детей (8%), так и с 1-4 детьми (8-9%). Случаев взятия кредитов людьми с 5 детьми недостаточно, чтобы делать какие-либо выводы (процент должников равен 0 по данной выборке). Таким образом, зависимости между количеством детей и возвратом кредита в срок нет. 

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Почему же нет? Разница между некоторыми группами (группами с 0 детей и с 1 ребенком) порядка 20%:
</div>    


In [102]:
# код ревьюера

(7.5 - 9.2) / 9.2

-0.1847826086956521

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Конечно, в идеале проверяют, является такая разница случайной или нет, используя стат. тесты и првоеряют к каким денежным потерям такая разница приводит, но здесь такой возможности нет. А так, такие доли это как раз то, что нужно. Можно определить, какая группа рисковая, а какая нет, найдя среднюю долю должников во всей выборке и сравнивать долю в каждой группе со средней.
    
Саму табличку можно оформить так:
</div>    


In [103]:
# Код ревьюера

data.groupby('children').agg(число_клиентов=('debt','count'),
                             число_должников=('debt','sum'),
                             доля_должников=('debt','mean')) \
                        .style.format({'доля_должников' : '{:.2%}'})

**Задание 20. Есть ли зависимость между семейным положением и возвратом кредита в срок?**

In [104]:
grouped_debt_family_status = data.groupby('family_status').agg({'debt':['sum','count']}) 
grouped_debt_family_status['percent_debt'] = grouped_debt_family_status['debt']['sum']/grouped_debt_family_status['debt']['count']*100 
print(grouped_debt_family_status)

                      debt        percent_debt
                       sum  count             
family_status                                 
Не женат / не замужем  273   2796     9.763948
в разводе               84   1189     7.064760
вдовец / вдова          63    951     6.624606
гражданский брак       385   4146     9.286059
женат / замужем        927  12266     7.557476


<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Чтобы не копировать один и тот же код, можно написать функцию и переиспользовать её</div>    


<div class = "alert alert-info" style="border-left: 7px solid blue">
<h3>✔ Комментарий студента 2 </h3>
Ниже прикрепляю код функции
</div>  

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера V.2</h3>
    
👍
</div>  

In [105]:
#код студента
def grouping(df, column):
    grouped_debt = df.groupby(column).agg({'debt':['sum','count']})
    grouped_debt['percent_debt'] = grouped_debt['debt']['sum']/grouped_debt['debt']['count']*100
    return grouped_debt

In [106]:
display(grouping(df=data, column='family_status'))
display(grouping(df=data, column='total_income_category'))
display(grouping(df=data, column='purpose_category'))

debt        percent_debt
                       sum  count             
family_status                                 
Не женат / не замужем  273   2796     9.763948
в разводе               84   1189     7.064760
вдовец / вдова          63    951     6.624606
гражданский брак       385   4146     9.286059
женат / замужем        927  12266     7.557476

debt        percent_debt
                        sum  count             
total_income_category                          
A                         2     25     8.000000
B                       354   5014     7.060231
C                      1353  15938     8.489145
D                        21    349     6.017192
E                         2     22     9.090909

debt        percent_debt
                          sum  count             
purpose_category                                 
операции с автомобилем    400   4281     9.343611
операции с недвижимостью  780  10754     7.253115
получение образования     369   3989     9.250439
проведение свадьбы        183   2324     7.874355

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера V.2</h3>
    
Ну вот, всего 3 строки :) 
</div>  

**Вывод:** Для женатых людей, людей в разводе и вдовцов/вдов процент должников составляет 7-8%, в то время как 10% не женатых и 9% людей, живущих в гражданском браке, не возвращают кредит в срок. 
Таким образом, можно утверждать, что существует зависимость между семейным положением и возвратом кредита в срок: люди, когда-либо состоящие в браке, реже попадают в ряды должников по выплатам кредита, чем те, что официально не женаты.

**Задание 21. Есть ли зависимость между уровнем дохода и возвратом кредита в срок?**

In [107]:
grouped_debt_total_income_category = data.groupby('total_income_category').agg({'debt':['sum','count']}) 
grouped_debt_total_income_category['percent_debt'] = grouped_debt_total_income_category['debt']['sum']/grouped_debt_total_income_category['debt']['count']*100 
print(grouped_debt_total_income_category)

                       debt        percent_debt
                        sum  count             
total_income_category                          
A                         2     25     8.000000
B                       354   5014     7.060231
C                      1353  15938     8.489145
D                        21    349     6.017192
E                         2     22     9.090909


**Вывод:** Логично было бы предположить, что чем ниже уровень дохода, тем выше вероятность возникновения задолженности. Процент должников с уровнем дохода категории "Е" (0-30000) составляет 9%, что является максимальным значением. Однако линейной зависимости между уровнем дохода и возвратом кредита в срок не наблюдается: минимальный процент должников имеет группа, с категорией дохода "D" (30001-50000) - 6%. Поэтому, прямой зависимости между уровнем дохода и возвратом кредита в срок нет. Для каждой группы процент задолженностей не превышает 10%.

**Задание 22. Как разные цели кредита влияют на его возврат в срок?**

In [108]:
grouped_debt_purpose_category = data.groupby('purpose_category').agg({'debt':['sum','count']}) 
grouped_debt_purpose_category['percent_debt'] = grouped_debt_purpose_category['debt']['sum']/grouped_debt_purpose_category['debt']['count']*100 
print(grouped_debt_purpose_category)

                         debt        percent_debt
                          sum  count             
purpose_category                                 
операции с автомобилем    400   4281     9.343611
операции с недвижимостью  780  10754     7.253115
получение образования     369   3989     9.250439
проведение свадьбы        183   2324     7.874355


**Вывод:** Для таких целей, как "операции с автомобилем" и "получение образования" наблюдается наибольший процент должников - 9.3%, в то время как при операциях с недвижимостью и проведении свадьбы задолженность возникает реже (7% и 8% соответственно).

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Выводить датафреймы, кстати, нужно через `display`, так они выводятся в `html` табличках</div>    


**Задание 23. Приведите возможные причины появления пропусков в исходных данных.**

*Ответ:* технологические (ошибки при вводе, копировании, считывании или форматировании), сокрытие информации 

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
На самом деле больший практический смысл имеет характер пропусков, а не сама причина. Тут пропуски распределены равномерно по датасету, что позволяет заменить их на медиану</div>    


**Задание 24. Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.**

*Ответ:* Среднее значение некорректно характеризует данные, когда некоторые значения сильно выделяются среди большинства. Медиана дает более объективную информацию о данных, чем среднее значение.

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
> Среднее значение некорректно характеризует данные

У такой формулировки очень широкий смысл. Медиана лучше описыват наиболее вероятные значения для выборки в случае, если есть выбивающиеся значения
</div>    


## Общий вывод.

По результатам исследования можно сделать вывод, что семейное положение влияет на факт погашения кредита в срок: люди, когда-либо состоящие в браке, реже попадают в ряды должников по выплатам кредита, чем те, что официально не женаты. Количество детей клиента не влияет на факт погашения кредита в срок, поскольку процент должников не превышает 10% как для людей без детей (8%), так и с 1-4 детьми (8-9%). 

<div class="alert alert-success"; style="border-left: 7px solid green">
<h3>✔ Комментарий ревьюера</h3>
    
Всё верно, можно прорезюмировать портреты добросовестного должника и недобросовестного.

Дальше в курсе будет статистика и научат опредлять, когда разница между группами случайная, а когда нет
</div>    
